ignition/boot-process.md

```diff
@@ -0,0 +1,37 @@
+# CoreOS startup process
+
+The CoreOS startup process is built on the standard [Linux startup process][linux startup]. Since this process is already well documented and generally well understood, this document will focus on aspects specific to booting CoreOS.
+
+## Bootloader
+
+[GRUB][grub] is the first program executed when a CoreOS system boots.
+
+The CoreOS [GRUB config][grubconfig] has several roles. First, the GRUB config [specifies which `usr` partition to use][gptprio.next] from the two `usr` partitions CoreOS uses to provide atomic upgrades and rollbacks.
+
+Additionally, GRUB determines if this is the first time a machine has booted. This is accomplished by [searching for the initial disk GUID][search] (00000000-0000-0000-0000-000000000001). CoreOS is built with this well-known disk GUID in order to detect the first boot. This GUID is randomized later in the boot process so that individual disks may be uniquely identified. If GRUB detects that this is in fact a first boot, it sets two Linux kernel command line parameters: `coreos.randomize_guid=00000000-0000-0000-0000-000000000001` and `coreos.first_boot=1`. These parameters are used by various programs in later stages of the boot process.
+
+## Early user space
+
+The next major milestone in the CoreOS startup process is the jump into the initial RAM file system. In addition to its standard responsibility of mounting the root filesystem, the initramfs is also where the disk GUID is randomized and where Ignition runs.
```

Comment:
The initrd and initramfs are actually different things. They serve the same purpose, but their semantics and format are different (initrd is legacy). If people recognize "initrd", I'm pretty sure they will recognize "initramfs" as well. Maybe not though.